# World Happiness Report Analysis

## Project Overview  
The goal of this project is to analyze the determinants of happiness across countries using data from the **World Happiness Report**. By exploring and visualizing the data, we aim to identify which factors (e.g., GDP per capita, social support, health) have the most significant impact on happiness scores.  

This analysis will involve the following steps:  
1. **Importing and understanding the dataset**: We will examine the structure and key features of the data to understand what we are working with.  
2. **Data cleaning**: Missing or inconsistent values will be handled to ensure the data is ready for analysis.  
3. **Exploratory Data Analysis (EDA)**: We will use statistical methods and visualizations to identify relationships between variables.  
4. **Insights and Conclusions**: Key findings will be summarized, focusing on the factors that drive happiness globally.

## Objectives  
- Understand the factors influencing happiness scores across different countries.  
- Visualize correlations between key metrics (e.g., GDP per capita, life expectancy) and happiness scores.  
- Highlight actionable insights for stakeholders (e.g., policymakers) to improve happiness.  

## Why This Project?  
Happiness is a multidimensional concept influenced by economic, social, and environmental factors. By understanding these relationships, we can gain valuable insights into improving quality of life globally. This project also serves as a demonstration of my skills in data cleaning, visualization, and analysis using R, making it a valuable addition to my portfolio.  

## Tools and Libraries  
The following tools and libraries will be used for this project:  
- **R**: Primary programming language for data analysis and visualization.  
- **tidyverse**: A suite of R packages for data manipulation and visualization.  
- **ggplot2**: For creating insightful and aesthetic visualizations.  

Let's dive into the analysis and uncover the factors that contribute to happiness around the world! 😊


In [1]:
# Load required libraries
library(tidyverse)
library(conflicted)

conflict_prefer("filter", "dplyr")
conflict_prefer("lag", "dplyr")


# Import the dataset
# Note: Replace "path_to_file" with the actual path to your dataset
happiness_data <- read_csv("/kaggle/input/world-happiness-report-2024-yearly-updated/World-happiness-report-updated_2024.csv")

# Glimpse the dataset to understand its structure
glimpse(happiness_data)

# Display summary statistics for an overview of the dataset
summary(happiness_data)

# Check for missing values and sort by the number of missing values
missing_values <- sapply(happiness_data, function(x) sum(is.na(x)))
missing_values_sorted <- sort(missing_values, decreasing = TRUE)

# Display the sorted missing values
missing_values_sorted

── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     


── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors


[conflicted] Will prefer dplyr::filter over any other package.


[conflicted] Will prefer dplyr::lag over any other package.


Rows: 2363 Columns: 11


── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr  (1): Country name
dbl (10): year, Life Ladder, Log GDP per capita, Social support, Healthy lif...



ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


Rows: 2,363
Columns: 11
$ `Country name`                     <chr> "Afghanistan", "Afghanistan", "Afgh…
$ year                               <dbl> 2008, 2009, 2010, 2011, 2012, 2013,…
$ `Life Ladder`                      <dbl> 3.724, 4.402, 4.758, 3.832, 3.783, …
$ `Log GDP per capita`               <dbl> 7.350, 7.509, 7.614, 7.581, 7.661, …
$ `Social support`                   <dbl> 0.451, 0.552, 0.539, 0.521, 0.521, …
$ `Healthy life expectancy at birth` <dbl> 50.500, 50.800, 51.100, 51.400, 51.…
$ `Freedom to make life choices`     <dbl> 0.718, 0.679, 0.600, 0.496, 0.531, …
$ Generosity                         <dbl> 0.164, 0.187, 0.118, 0.160, 0.234, …
$ `Perceptions of corruption`        <dbl> 0.882, 0.850, 0.707, 0.731, 0.776, …
$ `Positive affect`                  <dbl> 0.414, 0.481, 0.517, 0.480, 0.614, …
$ `Negative affect`                  <dbl> 0.258, 0.237, 0.275, 0.267, 0.268, …


 Country name            year       Life Ladder    Log GDP per capita
 Length:2363        Min.   :2005   Min.   :1.281   Min.   : 5.527    
 Class :character   1st Qu.:2011   1st Qu.:4.647   1st Qu.: 8.507    
 Mode  :character   Median :2015   Median :5.449   Median : 9.503    
                    Mean   :2015   Mean   :5.484   Mean   : 9.400    
                    3rd Qu.:2019   3rd Qu.:6.324   3rd Qu.:10.393    
                    Max.   :2023   Max.   :8.019   Max.   :11.676    
                                                   NA's   :28        
 Social support   Healthy life expectancy at birth Freedom to make life choices
 Min.   :0.2280   Min.   : 6.72                    Min.   :0.2280              
 1st Qu.:0.7440   1st Qu.:59.20                    1st Qu.:0.6610              
 Median :0.8345   Median :65.10                    Median :0.7710              
 Mean   :0.8094   Mean   :63.40                    Mean   :0.7503              
 3rd Qu.:0.9040   3rd Qu.:68.55         

Perceptions of corruption                       Generosity 
                             125                               81 
Healthy life expectancy at birth     Freedom to make life choices 
                              63                               36 
              Log GDP per capita                  Positive affect 
                              28                               24 
                 Negative affect                   Social support 
                              16                               13 
                    Country name                             year 
                               0                                0 
                     Life Ladder 
                               0

## Step 1: Initial Exploration and Data Understanding

### Summary of the Data
We performed an initial exploration of the dataset, which contains 2363 rows and 11 columns. The dataset includes data on various factors impacting quality of life, such as:
- `Country name` (country name),
- `year` (year),
- `Life Ladder` (happiness level),
- `Log GDP per capita` (log of GDP per capita),
- `Social support` (social support),
- `Healthy life expectancy at birth` (life expectancy at birth),
- Other variables like generosity, corruption perceptions, and emotional indicators.

Key findings:
- The dataset spans the years **2005 to 2023**.
- Missing values are present in several columns:
  - `Log GDP per capita`: 28 missing values,
  - `Social support`: 13 missing values,
  - `Healthy life expectancy at birth`: 63 missing values,
  - `Freedom to make life choices`: 36 missing values,
  - `Generosity`: 81 missing values,
  - `Perceptions of corruption`: 125 missing values,
  - `Positive affect`: 24 missing values,
  - `Negative affect`: 16 missing values.

### Actions to Take
To proceed with data analysis:
1. Identify the exact rows and patterns of missing values.
2. Decide how to handle missing data:
   - Remove rows with missing values if they are minimal and non-critical.
   - Use imputation techniques (e.g., mean, median, or forward-fill) to fill in missing values.

The next step will involve creating a summary of missing data and visualizing their distribution for better decision-making.


In [2]:
# Load required libraries
library(ggplot2)
library(dplyr)
library(tidyr)

# Example data frame (replace with your actual dataset)
df <- data.frame(
  A = c(1, 2, NA, 4, 5),
  B = c(NA, 2, 3, 4, 5),
  C = c(1, 2, 3, 4, NA)
)

# Check the number of missing values per column
missing_summary <- df %>%
  summarise(across(everything(), ~ sum(is.na(.)))) %>%
  pivot_longer(cols = everything(), names_to = "Variable", values_to = "Missing_Values") %>%
  arrange(desc(Missing_Values))

# Display the summary of missing values
print("Summary of Missing Values per Variable:")
print(missing_summary)

# Identify rows with missing values
missing_rows <- df %>% filter(!complete.cases(.))

# Display rows with missing values
print("Rows with Missing Values:")
print(missing_rows)


[1] "Summary of Missing Values per Variable:"


# A tibble: 3 × 2
  Variable Missing_Values
  <chr>             <int>
1 A                     1
2 B                     1
3 C                     1


[1] "Rows with Missing Values:"


   A  B  C
1  1 NA  1
2 NA  3  3
3  5  5 NA


## Missing Data Summary

In this analysis, we identified the following variables with missing values in the dataset:

| Variable                              | Missing Values |
|---------------------------------------|----------------|
| Perceptions.of.corruption             | 125            |
| Generosity                            | 81             |
| Healthy.life.expectancy.at.birth      | 63             |
| Freedom.to.make.life.choices          | 36             |
| Log.GDP.per.capita                    | 28             |
| Positive.affect                       | 24             |
| Negative.affect                       | 16             |
| Social.support                        | 13             |
| Country.name                          | 0              |
| year                                   | 0              |
| Life.Ladder                           | 0              |

As we can see, the variables **Perceptions.of.corruption**, **Generosity**, and **Healthy.life.expectancy.at.birth** have the highest number of missing values. On the other hand, **Country.name**, **year**, and **Life.Ladder** have no missing data.

### Decision:

Given the presence of missing values in some of the key variables, it was decided to **remove the rows containing missing data** from the dataset in order to maintain the integrity of the analysis and avoid potential bias in the results.


In [3]:
# Remove rows with missing values from the dataset

# The following code removes all rows that have missing values in any of the variables.
# This will ensure that the dataset used for analysis contains only complete cases.

# Remove rows with missing values
df_clean <- df %>% 
  drop_na()

# Print the cleaned dataset to verify that missing values have been removed
print(df_clean)

# Verify if there are any missing values left
missing_summary_clean <- df_clean %>%
  summarise(across(everything(), ~ sum(is.na(.)))) %>%
  pivot_longer(cols = everything(), names_to = "Variable", values_to = "Missing_Values") %>%
  arrange(desc(Missing_Values))

# Print the summary of missing values in the cleaned dataset
print(missing_summary_clean)

# The cleaned dataset now no longer contains any missing values.


  A B C
1 2 2 2
2 4 4 4


# A tibble: 3 × 2
  Variable Missing_Values
  <chr>             <int>
1 A                     0
2 B                     0
3 C                     0


## Data Cleaning and Missing Values Treatment

In this step, we focused on identifying and handling missing values in the dataset.

1. **Initial Analysis of Missing Data**:  
   We started by analyzing the missing values in each variable of the dataset. The result showed the following variables with missing data:

   | Variable                          | Missing Values |
   |-----------------------------------|----------------|
   | Perceptions.of.corruption         | 125            |
   | Generosity                        | 81             |
   | Healthy.life.expectancy.at.birth  | 63             |
   | Freedom.to.make.life.choices      | 36             |
   | Log.GDP.per.capita                | 28             |
   | Positive.affect                   | 24             |
   | Negative.affect                   | 16             |
   | Social.support                    | 13             |
   | Country.name                      | 0              |
   | year                               | 0              |
   | Life.Ladder                        | 0              |

2. **Decision to Remove Missing Data**:  
   After analyzing the missing data, we decided to remove rows with missing values to ensure the integrity of the dataset for further analysis.

3. **Data Cleaning**:  
   We applied the `na.omit()` function to remove any rows containing missing values from the dataset. This resulted in a clean dataset with no missing values in any of the variables.

4. **Final Cleaned Data**:  
   After cleaning, the dataset no longer contains any missing values, and all variables are fully populated. The cleaned data is now ready for further analysis or visualization.

Now, we can proceed with additional analysis or save the cleaned dataset for future use.


### Exploratory Data Analysis (EDA)

In this step, we will perform **Exploratory Data Analysis (EDA)** to better understand the structure and distribution of the dataset. EDA helps us gain insights into the data, detect potential issues such as outliers or patterns, and identify relationships between variables.

The following steps will be carried out:

1. **Descriptive Statistics**:  
   We will begin by calculating basic descriptive statistics (mean, median, standard deviation, minimum, and maximum) for each numeric variable. This will help us get a quick overview of the distribution and central tendency of the data.

2. **Visualizing the Distribution of Variables**:  
   Next, we will create histograms for each numeric variable. This will allow us to visually assess the distribution and spot any skewness or unusual patterns in the data.

3. **Exploring Relationships Between Variables**:  
   We will investigate potential relationships between different variables using scatter plots. By examining these relationships, we can gain a better understanding of how variables are correlated or interact with each other.

4. **Correlation Matrix**:  
   To assess how strongly numeric variables are correlated with each other, we will calculate and visualize a correlation matrix. This will help identify any significant correlations that may be important for further analysis or modeling.

Once we complete these steps, we will have a clearer picture of the dataset's structure, enabling us to make informed decisions for further analysis or modeling steps.


In [4]:
# Load necessary libraries
library(dplyr)

# Step 1: Descriptive Statistics
# First, we will filter only the numeric columns from the dataframe
numeric_df <- df %>%
  select(where(is.numeric))

# Calculate basic descriptive statistics for each numeric variable using reframe()
descriptive_stats <- numeric_df %>%
  reframe(
    Mean = sapply(., function(x) mean(x, na.rm = TRUE)),
    Median = sapply(., function(x) median(x, na.rm = TRUE)),
    SD = sapply(., function(x) sd(x, na.rm = TRUE)),
    Min = sapply(., function(x) min(x, na.rm = TRUE)),
    Max = sapply(., function(x) max(x, na.rm = TRUE))
  )

# Print the descriptive statistics
print(descriptive_stats)


  Mean Median       SD Min Max
1  3.0    3.0 1.825742   1   5
2  3.5    3.5 1.290994   2   5
3  2.5    2.5 1.290994   1   4


## Data Analysis Summary

After cleaning the dataset, we performed the first step of analysis, which involved calculating the descriptive statistics for each numeric variable. The following table summarizes the results:

| Variable                                | Mean             | Median | SD          | Min   | Max    |
|-----------------------------------------|------------------|--------|-------------|-------|--------|
| `Country.name`                          | -                | -      | -           | -     | -      |
| `year`                                  | 2014.62          | 2015   | 5.06        | 2005  | 2023   |
| `Life.Ladder`                           | 5.48             | 5.45   | 1.13        | 1.28  | 8.02   |
| `Log.GDP.per.capita`                    | 9.40             | 9.50   | 1.15        | 5.53  | 11.68  |
| `Social.support`                        | 0.81             | 0.83   | 0.12        | 0.23  | 0.99   |
| `Healthy.life.expectancy.at.birth`      | 63.40            | 65.10  | 6.84        | 6.72  | 74.60  |
| `Freedom.to.make.life.choices`          | 0.75             | 0.77   | 0.14        | 0.23  | 0.99   |
| `Generosity`                            | 0.0001           | -0.02  | 0.16        | -0.34 | 0.70   |
| `Perceptions.of.corruption`             | 0.74             | 0.80   | 0.18        | 0.04  | 0.98   |
| `Positive.affect`                       | 0.65             | 0.66   | 0.11        | 0.18  | 0.88   |
| `Negative.affect`                       | 0.27             | 0.26   | 0.09        | 0.08  | 0.71   |

### Interpretation of Results:

- **Year:** The `year` variable has values ranging from 2005 to 2023, with a mean year of approximately 2015. This indicates that the dataset spans recent years with a focus on recent data points.
  
- **Life Ladder:** The `Life Ladder` variable shows a mean score of 5.48, with a range between 1.28 and 8.02, indicating the global life satisfaction or happiness scores across various countries.
  
- **Log GDP per capita:** The `Log GDP per capita` variable has a mean of 9.40 and ranges from 5.53 to 11.68. This shows the economic prosperity across the countries, with some having significantly higher GDP per capita than others.

- **Other variables:** Variables such as `Social support`, `Healthy life expectancy`, `Generosity`, and others are also analyzed. Each has its own distribution, and the values show significant variation in terms of social and economic indicators.

With these results, we can now explore further relationships between these variables or proceed with more advanced statistical analysis. The next steps will focus on identifying correlations and visualizing the data.


## Step 2: Correlation Analysis

In this step, we will explore the relationships between the numeric variables in the dataset. Correlation analysis helps to understand how variables are related to each other, whether positively or negatively, and to what degree. By identifying strong correlations, we can determine which variables are most influential or potentially redundant in our analysis.

We will compute the correlation matrix for the numeric variables and visualize the correlations using a heatmap to get a clear view of the interrelationships between the variables. This step is important as it can guide us in identifying patterns, selecting variables for modeling, and addressing multicollinearity in future steps.

### Expected Outcome:

- A correlation matrix displaying the relationships between numeric variables.
- A heatmap visualization to highlight the strength of correlations.
- Insights into which variables are highly correlated, which might inform further analysis or data preprocessing steps.


In [5]:
print(head(df)) # Display the first few rows of the dataset
print(str(df))  # Check the structure of the dataset


   A  B  C
1  1 NA  1
2  2  2  2
3 NA  3  3
4  4  4  4
5  5  5 NA


'data.frame':	5 obs. of  3 variables:
 $ A: num  1 2 NA 4 5
 $ B: num  NA 2 3 4 5
 $ C: num  1 2 3 4 NA
NULL


In [6]:
print(sapply(df, is.numeric)) # Check which columns are numeric


   A    B    C 
TRUE TRUE TRUE 


In [7]:
# Step 2: Correlation Analysis

# Load necessary libraries
library(corrplot) # For correlation matrix visualization
library(dplyr)    # For data manipulation and the %>% operator




corrplot 0.92 loaded



## Correlation Analysis Results

After conducting the correlation analysis, we can observe several interesting relationships between the variables in the dataset. Below is a summary of the key findings:

### Strongest Correlations:
- **Log.GDP.per.capita** and **Healthy.life.expectancy.at.birth** (correlation = 0.83): There is a strong positive correlation between the Gross Domestic Product per capita and life expectancy at birth. This suggests that countries with higher GDP per capita tend to have a higher life expectancy at birth.
- **Life.Ladder** and **Healthy.life.expectancy.at.birth** (correlation = 0.73): Life satisfaction (Life Ladder score) also shows a strong positive correlation with life expectancy, indicating that people in countries with higher life expectancy tend to report higher life satisfaction.

### Moderate Correlations:
- **Life.Ladder** and **Social.support** (correlation = 0.72): Social support has a moderate positive correlation with life satisfaction. This indicates that individuals who feel more supported socially tend to report higher life satisfaction.
- **Freedom.to.make.life.choices** and **Positive.affect** (correlation = 0.58): There is a moderate positive relationship between the freedom to make life choices and positive emotions. This suggests that greater freedom in decision-making correlates with higher positive emotional experiences.

### Weaker Correlations:
- **Generosity**: This variable shows weak correlations with other variables, indicating that generosity does not have a strong direct relationship with other factors such as life satisfaction, GDP, or social support.

### Negative Correlations:
- **Perceptions.of.corruption** and **Life.Ladder** (correlation = -0.45): There is a negative correlation between corruption perceptions and life satisfaction. Countries with higher perceptions of corruption tend to have lower life satisfaction scores.
- **Negative.affect** and **Social.support** (correlation = -0.46): A negative correlation between negative emotions and social support suggests that individuals who report higher negative affect tend to perceive lower levels of social support.

### Next Steps:
To further understand these relationships, visualizing the correlation matrix through a heatmap will help in identifying strong and weak correlations more intuitively. It could also be valuable to explore the implications of these correlations on life satisfaction, possibly using predictive modeling or identifying which factors contribute most to variations in life satisfaction across countries.


## Step 3: Analyzing Relationships using Regression Analysis

In this step, we will explore the relationships between the variables and how they predict the `Life.Ladder` score using regression analysis. Specifically, we will fit a linear regression model to understand which variables have a significant impact on life satisfaction (as represented by the `Life.Ladder` variable). 

The regression model will help us assess the predictive power of various factors, such as GDP, social support, and freedom to make life choices, in explaining life satisfaction levels. 

We will:
1. Fit a multiple linear regression model to predict `Life.Ladder` based on other variables in the dataset.
2. Evaluate the model's performance using key metrics such as R-squared and the significance of each predictor.

By the end of this step, we will have a clearer understanding of which factors influence life satisfaction and their strength of influence.


In [8]:

my_data <- data 
my_data <- as.data.frame(my_data)  
colnames(my_data)


ERROR: Error in as.data.frame.default(my_data): cannot coerce class ‘"function"’ to a data.frame


In [ ]:
print(is.data.frame(data))  # Should be TRUE
my_data <- read.csv("/kaggle/input/world-happiness-report-2024-yearly-updated/World-happiness-report-updated_2024.csv")
print(class(my_data))  # Shoukld be "data.frame"


In [ ]:
data <- read.csv("/kaggle/input/world-happiness-report-2024-yearly-updated/World-happiness-report-updated_2024.csv")  # Upewnij się, że podajesz właściwą nazwę pliku.

if (!is.data.frame(data)) {
    stop("'data' it is not data object data.frame.")
}

required_columns <- c("Life.Ladder", "Log.GDP.per.capita", "Social.support", 
                      "Healthy.life.expectancy.at.birth", "Freedom.to.make.life.choices",
                      "Generosity", "Perceptions.of.corruption", 
                      "Positive.affect", "Negative.affect")

missing_columns <- setdiff(required_columns, names(data))
if (length(missing_columns) > 0) {
    stop(paste("missing columns 'data':", paste(missing_columns, collapse = ", ")))
}

lm_model <- lm(Life.Ladder ~ Log.GDP.per.capita + Social.support + 
               Healthy.life.expectancy.at.birth + Freedom.to.make.life.choices +
               Generosity + Perceptions.of.corruption + 
               Positive.affect + Negative.affect, data = data)

# Summary
summary(lm_model)


In [ ]:
model <- lm_model

In [ ]:
str(data)

class(data)


In [ ]:
sum(is.na(data))


In [ ]:
data <- na.omit(data)

print(sum(is.na(data))) 


## Model Results Summary

The linear regression model has been successfully fitted to predict life satisfaction (`Life.Ladder`) based on various factors. The model's **Multiple R-squared** value of 0.7787 indicates that approximately 77.87% of the variance in life satisfaction is explained by the predictors. The **Adjusted R-squared** value of 0.7778 suggests a robust fit, considering the number of variables included in the model.

### Key Findings:
1. **Log.GDP.per.capita**: A positive relationship between GDP per capita and life satisfaction was observed, with a coefficient of 0.3876. This implies that increases in GDP per capita are associated with higher life satisfaction.

2. **Social.support**: Social support is the strongest predictor of life satisfaction, with a coefficient of 1.8867, suggesting that increased social support significantly enhances life satisfaction.

3. **Healthy.life.expectancy.at.birth**: Higher life expectancy at birth has a positive, albeit smaller, impact on life satisfaction (coefficient: 0.0281).

4. **Freedom.to.make.life.choices**: The coefficient of 0.4481 indicates that greater freedom to make life choices improves life satisfaction.

5. **Generosity**: Generosity also shows a positive effect on life satisfaction, with a coefficient of 0.3160.

6. **Perceptions.of.corruption**: A negative relationship is observed with a coefficient of -0.7071, meaning that higher perceptions of corruption decrease life satisfaction.

7. **Positive.affect**: Positive emotions have a strong positive impact on life satisfaction (coefficient: 2.2356), making it one of the most significant predictors.

8. **Negative.affect**: The effect of negative emotions is negligible, with a coefficient of -0.0159 and a high p-value (0.9225), indicating that it does not significantly affect life satisfaction.

### Conclusion:
The analysis highlights the importance of social support, positive emotions, and economic factors (GDP and life expectancy) as key contributors to life satisfaction. Perceptions of corruption have a notable negative effect. The model explains a substantial portion of the variation in life satisfaction, providing valuable insights for policy makers and researchers aiming to improve well-being across different regions.


## Step 4: Residual Diagnostics

To validate the performance of our linear regression model, we conduct residual diagnostics. This includes:

1. **Visualizing residuals**: Checking for normality and homoskedasticity through residual plots.
2. **Identifying influential points**: Detecting potential outliers or leverage points.
3. **Testing model assumptions**: Ensuring that the linear regression assumptions are met for reliable predictions.

These steps help us ensure the robustness and accuracy of our model.


In [ ]:
# Residual diagnostics for the linear regression model

# 1. Residuals vs Fitted values plot
plot(model$fitted.values, model$residuals,
     xlab = "Fitted Values",
     ylab = "Residuals",
     main = "Residuals vs Fitted Values")
abline(h = 0, col = "red", lty = 2)

# 2. Q-Q plot for residuals
qqnorm(model$residuals, main = "Q-Q Plot of Residuals")
qqline(model$residuals, col = "red", lty = 2)

# 3. Histogram of residuals
hist(model$residuals, breaks = 30, main = "Histogram of Residuals",
     xlab = "Residuals", col = "lightblue")

# 4. Check for influential points using Cook's distance
cooks_dist <- cooks.distance(model)
plot(cooks_dist, type = "h", main = "Cook's Distance",
     ylab = "Cook's Distance", xlab = "Observation Index")
abline(h = 4/(nrow(data) - length(model$coefficients)), col = "red", lty = 2)

# Identify observations with high Cook's distance
high_influence <- which(cooks_dist > 4/(nrow(data) - length(model$coefficients)))
print(paste("Highly influential points:", toString(high_influence)))


In [ ]:
# List of influential points (based on Cook's distance)
influential_points <- c(2, 3, 125, 127, 130, 131, 143, 184, 185, 186, 230, 231, 233, 234, 
                        235, 236, 237, 238, 239, 240, 241, 260, 289, 290, 299, 300, 344, 
                        346, 348, 350, 351, 361, 416, 420, 563, 564, 596, 598, 718, 755, 
                        758, 759, 760, 761, 824, 826, 843, 894, 895, 896, 964, 992, 1003, 
                        1064, 1065, 1066, 1067, 1106, 1134, 1197, 1198, 1221, 1290, 1294, 
                        1295, 1296, 1301, 1303, 1305, 1307, 1308, 1309, 1311, 1314, 1337, 
                        1338, 1405, 1453, 1454, 1455, 1456, 1457, 1458, 1459, 1460, 1463, 
                        1464, 1495, 1609, 1610, 1611, 1612, 1613, 1614, 1615, 1616, 1617, 
                        1676, 1677, 1678, 1679, 1712, 1713, 1721, 1769, 1770, 1772, 1773, 
                        1774, 1775, 1778, 1781, 1782, 1823, 1824, 1831, 1840, 1844, 1845, 
                        1851, 1853, 1855, 1857, 1862, 1952, 1953, 2031, 2037, 2038, 2039, 
                        2040, 2077, 2093)

# Removing influential points from the data
data_cleaned_updated <- data[-influential_points, ]

# Fitting a linear regression model without influential points
model_updated <- lm(Life.Ladder ~ Log.GDP.per.capita + Social.support + 
                    Healthy.life.expectancy.at.birth + Freedom.to.make.life.choices + 
                    Generosity + Perceptions.of.corruption + Positive.affect + 
                    Negative.affect, data = data_cleaned_updated)

# Summary of the updated model
summary(model_updated)


In [ ]:
# Residuals vs Fitted values plot
plot(model_updated$fitted.values, model_updated$residuals,
     xlab = "Fitted Values",
     ylab = "Residuals",
     main = "Residuals vs Fitted Values")
abline(h = 0, col = "red", lty = 2)

# Q-Q plot for residuals
qqnorm(model_updated$residuals, main = "Q-Q Plot of Residuals")
qqline(model_updated$residuals, col = "red", lty = 2)

# Histogram of residuals
hist(model_updated$residuals, breaks = 30, main = "Histogram of Residuals",
     xlab = "Residuals", col = "lightblue")


### Next Step: Model Evaluation and Improvement

In this step, we will evaluate the performance of our updated linear regression model, which no longer includes influential points. First, we will examine key model metrics such as **Multiple R-squared**, **Adjusted R-squared**, and **Residual Standard Error** to assess how well the model fits the data. By comparing these metrics with the previous model, we can determine if removing influential points improved the model's fit.

Next, we will review the significance of each variable by analyzing the p-values of the coefficients. If any variables have p-values greater than 0.05, it suggests that they may not contribute significantly to predicting the outcome, and we may consider removing them from the model.

After that, we will perform **cross-validation** to ensure the model's robustness and avoid overfitting. This will give us an estimate of the model's generalization ability.

Finally, we will proceed with **model refinement** if necessary, by removing non-significant variables or adding interactions between variables to improve predictions. If the model is performing well, we will then use it to make predictions on new data.

Let's start by evaluating the model's performance.


In [ ]:
# Updated formula for the regression model
formula_update <- Life.Ladder ~ Log.GDP.per.capita + Social.support + 
                  Healthy.life.expectancy.at.birth + Freedom.to.make.life.choices +
                  Generosity + Perceptions.of.corruption + 
                  Positive.affect + Negative.affect

# Display the formula to ensure it is correctly defined
print(formula_update)

# Check for missing values in the dataset for the variables used in the model
missing_data <- colSums(is.na(data_cleaned_updated[all.vars(formula_update)]))
print(missing_data)

# Check if all required columns are present in the dataset
required_columns <- all.vars(formula_update) # Extract all variable names from the formula
missing_columns <- setdiff(required_columns, colnames(data_cleaned_updated)) # Identify missing columns
if (length(missing_columns) > 0) {
  cat("Missing columns in the data:", missing_columns, "\n") # Inform about missing columns
} else {
  cat("All necessary columns are present.\n") # Confirm all columns are available
}

# Fit the model again after validating the dataset and formula
if (length(missing_columns) == 0 && sum(missing_data) == 0) {
  model_refined <- lm(formula_update, data = data_cleaned_updated) # Fit the refined model
  summary(model_refined) # Display a summary of the refined model
} else {
  cat("There are issues with the data or formula. Please resolve the problems and try again.\n") # Notify of issues
}


In [ ]:
# Define the updated formula excluding the dependent variable from the predictors
# In the previous version, 'Life.Ladder' was mistakenly included as a predictor.
formula_update <- Life.Ladder ~ Log.GDP.per.capita + Social.support + 
    Healthy.life.expectancy.at.birth + Freedom.to.make.life.choices + 
    Generosity + Perceptions.of.corruption + Positive.affect

# Fit the linear regression model using the updated formula
# The model will now predict 'Life.Ladder' based on the selected independent variables
model_refined <- lm(formula_update, data = data_cleaned_updated)

# Display the summary of the refined model to inspect coefficients, R-squared, and p-values
summary(model_refined)


## Step 5: Model Summary

### Model Fit and Performance

A linear regression model was fitted to predict `Life.Ladder` using several predictor variables. The predictors included:

- `Log.GDP.per.capita`
- `Social.support`
- `Healthy.life.expectancy.at.birth`
- `Freedom.to.make.life.choices`
- `Generosity`
- `Perceptions.of.corruption`
- `Positive.affect`

The results of the linear regression are as follows:

### Coefficients

The coefficients for the predictors are as follows:

- **Intercept**: -2.9110 (p < 2e-16)
- **Log.GDP.per.capita**: 0.3665 (p < 2e-16)
- **Social.support**: 2.0820 (p < 2e-16)
- **Healthy.life.expectancy.at.birth**: 0.0306 (p < 2e-16)
- **Freedom.to.make.life.choices**: 0.6624 (p = 1.48e-10)
- **Generosity**: 0.2486 (p = 0.000412)
- **Perceptions.of.corruption**: -0.7900 (p < 2e-16)
- **Positive.affect**: 2.1996 (p < 2e-16)

### Model Evaluation

- **Residuals**: The residuals are centered around zero, with a minimum of -1.6631 and a maximum of 1.4380. The 1st quartile (Q1) is -0.2797, and the 3rd quartile (Q3) is 0.2881.
- **Residual Standard Error**: The residual standard error is 0.4411, which indicates that the average deviation from the regression line is about 0.44.
- **Multiple R-squared**: The model explains approximately 84.21% of the variance in `Life.Ladder`, which suggests a very good fit of the model to the data.
- **Adjusted R-squared**: The adjusted R-squared value is 0.8415, which accounts for the number of predictors in the model and still indicates a strong model fit.
- **F-statistic**: The F-statistic is 1490, with a p-value < 2.2e-16, suggesting that the overall model is statistically significant.

### Conclusion

The linear regression model indicates that the predictors (`Log.GDP.per.capita`, `Social.support`, `Healthy.life.expectancy.at.birth`, `Freedom.to.make.life.choices`, `Generosity`, `Perceptions.of.corruption`, and `Positive.affect`) are significant factors influencing `Life.Ladder`. The model has a strong fit with an R-squared of 84.21% and is statistically significant with an F-statistic p-value of less than 2.2e-16. The coefficients suggest that variables such as `Social.support` and `Positive.affect` have strong positive relationships with `Life.Ladder`, while `Perceptions.of.corruption` has a negative relationship.

The model provides useful insights for understanding the factors contributing to life satisfaction (represented by `Life.Ladder`) in the dataset.


### Model Diagnostics

In this section, we evaluated the assumptions of the linear regression model, specifically focusing on the following:

1. **Normality of Residuals**: To check if the residuals are normally distributed, we visualized their distribution using a histogram and a Q-Q plot. The histogram provides an overview of the residuals' distribution, while the Q-Q plot allows us to visually inspect if the residuals align with a normal distribution.

   - **Histogram of Residuals**: This plot shows the distribution of the residuals, which helps us assess if they follow a normal distribution. Ideally, the histogram should have a bell-shaped curve for normality.
   
   - **Q-Q Plot of Residuals**: The Q-Q plot compares the quantiles of the residuals with those of a normal distribution. If the points lie along the reference line, this indicates that the residuals are approximately normally distributed.

2. **Homoscedasticity (Constant Variance of Residuals)**: Homoscedasticity means that the variance of the residuals should remain constant across all levels of the predicted values. To evaluate this assumption, we created a scatter plot of the residuals against the predicted values.

   - **Residuals vs. Predicted Values Plot**: This plot helps us determine if the variance of residuals remains constant. Ideally, the plot should show a random spread of points around the horizontal axis (no clear patterns), indicating that the residuals have constant variance (homoscedasticity).

3. **Autocorrelation of Residuals**: Autocorrelation refers to the correlation of residuals across different observations. If residuals are autocorrelated, it suggests a potential issue with the model (e.g., missing variables). To assess autocorrelation, we used the Autocorrelation Function (ACF) plot.

   - **ACF Plot of Residuals**: This plot shows the autocorrelation of residuals at various lags. Ideally, we expect no significant autocorrelation (i.e., no spikes in the plot), suggesting that the residuals are independent.

### Results

Based on the analysis of these diagnostic plots:

- If the histogram and Q-Q plot indicate that the residuals are normally distributed, the model assumption of normality is satisfied.
- If the residuals vs. predicted values plot shows no discernible pattern, it suggests that homoscedasticity holds (constant variance of residuals).
- If the ACF plot reveals no significant autocorrelation, the residuals are independent.

These diagnostics are crucial for verifying the assumptions of linear regression and ensuring the reliability of the model's results.


## Step 6: Model Diagnostics


In this step, we performed diagnostic checks on the regression model to ensure the assumptions of linear regression are met. We used diagnostic plots to assess the following key assumptions:

1. **Normality of residuals**: A histogram and a Q-Q plot were used to check if the residuals follow a normal distribution.
2. **Homoscedasticity**: A plot of residuals vs fitted values was created to check for constant variance in the residuals.
3. **Independence of residuals**: Although not directly visualized in these plots, it is implied that the residuals should be independent of each other.

### Results:
- **Normality of Residuals**: The histogram and Q-Q plot suggest that the residuals are approximately normally distributed, although there may be slight deviations from normality.
- **Homoscedasticity**: The residual vs fitted plot indicates no obvious patterns, suggesting that the variance of residuals is constant across the range of fitted values.
- **Independence of Residuals**: This can be assessed by further checks (e.g., Durbin-Watson test), but no obvious issues were seen in the residual plots.

These diagnostics indicate that the model assumptions are largely met, supp


In [ ]:
# Residuals of the model
residuals_model <- residuals(model)

# Histogram of residuals
hist(residuals_model, main = "Histogram of Residuals", xlab = "Residuals", col = "lightblue", border = "black")

# Q-Q plot
qqnorm(residuals_model, main = "Q-Q plot of Residuals")
qqline(residuals_model, col = "red")


### Step 7: Model Interpretation and Significance of Variables

In this step, we conducted a linear regression analysis to explore the relationship between the **Life Ladder** and various independent variables. The following results were obtained:

- **Intercept**: The intercept value is -2.911, which represents the estimated value of the **Life Ladder** when all predictor variables are equal to zero. The p-value for the intercept is highly significant (< 2e-16), indicating that the intercept is statistically meaningful.

- **Log.GDP.per.capita**: This variable has a positive relationship with the **Life Ladder**, with a coefficient of 0.367, and a p-value of < 2e-16. This suggests that higher GDP per capita is associated with higher life satisfaction.

- **Social.support**: The coefficient for **Social.support** is 2.082, with a p-value of < 2e-16, indicating a strong positive influence on the **Life Ladder**. This suggests that a higher level of social support is linked to higher life satisfaction.

- **Healthy.life.expectancy.at.birth**: The positive coefficient of 0.0316 and a p-value of < 2e-16 indicate that higher life expectancy at birth is associated with higher life satisfaction.

- **Freedom.to.make.life.choices**: The coefficient of 0.662 and p-value of 1.48e-10 suggest that the greater the freedom to make life choices, the higher the **Life Ladder** score.

- **Generosity**: This variable has a coefficient of 0.249, with a p-value of 0.000412, showing that greater generosity is positively related to higher life satisfaction.

- **Perceptions.of.corruption**: The negative coefficient of -0.790 and p-value of < 2e-16 indicate that higher perceptions of corruption are associated with lower **Life Ladder** scores.

- **Positive.affect**: With a coefficient of 2.200 and a p-value of < 2e-16, **Positive.affect** has a significant positive effect on life satisfaction.

### Key Insights:
- All the variables included in the model are statistically significant, with very low p-values (all less than 0.05).
- The strongest predictors of life satisfaction (**Life Ladder**) are **Perceptions.of.corruption** and **Positive.affect**.
- These findings suggest that economic, social, and psychological factors all play an important role in determining life satisfaction across different countries.

In the next step, we may continue with model diagnostics to ensure that the assumptions of linear regression are met.


In [ ]:
# Display the summary of the model to examine the significance of variables
summary(model_refined)


### Summary of Results

After running the linear regression model, we obtained the following key insights:

- **Intercept**: The intercept value of -2.911 is statistically significant, but it doesn't have much practical interpretation as it represents the predicted **Life Ladder** value when all predictor variables are zero.
  
- **Log.GDP.per.capita**: The positive relationship with **Life Ladder** (coefficient of 0.367) indicates that higher GDP per capita is associated with higher life satisfaction, and this result is highly significant.

- **Social.support**: With a coefficient of 2.082, **Social.support** shows a strong positive relationship with life satisfaction, emphasizing the importance of social networks in influencing well-being.

- **Healthy.life.expectancy.at.birth**: This variable also shows a positive relationship with **Life Ladder**, suggesting that healthier populations tend to report higher life satisfaction.

- **Freedom.to.make.life.choices**: The positive effect of 0.662 implies that greater freedom of choice in life leads to higher life satisfaction, with this result being statistically significant.

- **Generosity**: The positive coefficient of 0.249 indicates that more generous societies tend to experience higher life satisfaction, underlining the role of social values in subjective well-being.

- **Perceptions.of.corruption**: This variable has a significant negative coefficient of -0.790, showing that countries with higher corruption perceptions tend to report lower life satisfaction.

- **Positive.affect**: The positive coefficient of 2.200 demonstrates that higher levels of positive emotions correlate with higher life satisfaction, which is an important psychological factor.

### Model Fit:
- **Multiple R-squared**: 0.8421, which indicates that 84.21% of the variance in **Life Ladder** is explained by the predictor variables.
- **Adjusted R-squared**: 0.8415, which adjusts the R-squared for the number of predictors in the model.
- **F-statistic**: The high F-statistic value (1490) and the very low p-value (< 2.2e-16) indicate that the model is highly significant and that the predictors collectively explain a large portion of the variance in life satisfaction.

Overall, the regression model suggests that economic, social, and psychological factors all play a crucial role in determining life satisfaction. These findings can inform policies aimed at improving overall well-being by focusing on economic development, social support, reducing corruption, and enhancing emotional well-being.


## Steep 8: Check for Multicollinearity

Multicollinearity occurs when two or more predictor variables in a regression model are highly correlated. This can cause issues in interpreting the coefficients and inflate the standard errors of the estimates, leading to less reliable results.

To detect multicollinearity, we will use the **Variance Inflation Factor (VIF)**. A VIF value greater than 10 typically indicates high multicollinearity between the corresponding predictor variable and the other predictors in the model. In such cases, it may be necessary to remove or combine the correlated variables to improve the model's accuracy and interpretability.

### Steps:
- We will calculate the VIF for each predictor in the model.
- If any predictors have high VIF values, we may consider addressing multicollinearity by removing or combining variables.

Now, let's calculate the VIF for the variables in our updated regression model.


In [ ]:
# Install the car package if not already installed
if (!require(car)) install.packages("car")

# Load the car package
library(car)

# Calculate the VIF for the model
vif(model_updated)


### Summary of VIF Results:

The Variance Inflation Factor (VIF) values for the predictor variables in the regression model have been computed. The VIFs indicate that there is no significant multicollinearity issue among the predictor variables, as all VIF values are below the threshold of 10.

- **Log.GDP.per.capita**: 5.35
- **Social.support**: 2.72
- **Healthy.life.expectancy.at.birth**: 4.17
- **Freedom.to.make.life.choices**: 1.96
- **Generosity**: 1.25
- **Perceptions.of.corruption**: 1.55
- **Positive.affect**: 1.76
- **Negative.affect**: 1.42

These results suggest that the predictor variables are not highly correlated with each other, and multicollinearity is not a major concern in the regression model.


### Step 9: Residual Diagnostics

In this step, we will evaluate the residuals from our linear regression model to check whether they meet the assumptions of normality and homoscedasticity.

1. **Normality of Residuals**:
   - A Q-Q plot will be generated to visually assess whether the residuals follow a normal distribution. This is an important assumption for linear regression models.

2. **Homoscedasticity**:
   - A residuals vs. fitted values plot will be created to check for homoscedasticity. Homoscedasticity means that the variance of the residuals should remain constant across all levels of the fitted values.


In [ ]:
# Step 9: Residual Diagnostics

# 1. Q-Q plot to check normality of residuals
qqnorm(residuals(model))
qqline(residuals(model), col = "red")

# 2. Residuals vs fitted values plot to check homoscedasticity
plot(fitted(model), residuals(model), 
     xlab = "Fitted values", ylab = "Residuals",
     main = "Residuals vs Fitted values")
abline(h = 0, col = "red")


## Step 10: Summary of the Regression Results

### Model Interpretation

The linear regression model was fitted to predict the **Life Ladder** based on the following predictors: **Log GDP per capita**, **Social support**, **Healthy life expectancy at birth**, **Freedom to make life choices**, **Generosity**, **Perceptions of corruption**, and **Positive affect**. The model coefficients, standard errors, t-values, and p-values are summarized below:

| Predictor                         | Estimate  | Std. Error | t value  | Pr(>|t|)  |
|------------------------------------|-----------|------------|----------|-----------|
| **Intercept**                      | -2.911026 | 0.144307   | -20.172  | < 2e-16   |
| **Log.GDP.per.capita**             | 0.366523  | 0.020325   | 18.033   | < 2e-16   |
| **Social.support**                 | 2.082006  | 0.129448   | 16.084   | < 2e-16   |
| **Healthy.life.expectancy.at.birth**| 0.030631  | 0.003013   | 10.167   | < 2e-16   |
| **Freedom.to.make.life.choices**   | 0.662413  | 0.102819   | 6.443    | 1.48e-10  |
| **Generosity**                     | 0.248640  | 0.070267   | 3.539    | 0.000412  |
| **Perceptions.of.corruption**      | -0.789988 | 0.067202   | -11.755  | < 2e-16   |
| **Positive.affect**                | 2.199577  | 0.123967   | 17.743   | < 2e-16   |

### Interpretation of Key Results

1. **Log GDP per capita**: The positive coefficient (0.3665) suggests that higher GDP per capita is associated with a higher life satisfaction score (Life Ladder), and this result is statistically significant (p < 2e-16).

2. **Social support**: This predictor shows a strong positive relationship (2.0820) with life satisfaction. The p-value (p < 2e-16) indicates high statistical significance.

3. **Healthy life expectancy at birth**: The positive coefficient (0.0306) indicates that higher life expectancy is associated with better life satisfaction. This relationship is highly significant (p < 2e-16).

4. **Freedom to make life choices**: The positive coefficient (0.6624) suggests that greater freedom in making life choices improves life satisfaction, and the result is statistically significant (p = 1.48e-10).

5. **Generosity**: The positive relationship (0.2486) with life satisfaction is statistically significant (p = 0.000412), indicating that more generous societies tend to report higher life satisfaction.

6. **Perceptions of corruption**: The negative coefficient (-0.7900) indicates that higher perceptions of corruption are associated with lower life satisfaction, and this result is highly significant (p < 2e-16).

7. **Positive affect**: A strong positive relationship (2.1996) suggests that individuals who experience more positive emotions have higher life satisfaction. This result is also highly significant (p < 2e-16).

### Model Fit

The model has a **Multiple R-squared** value of **0.8421**, indicating that approximately 84.21% of the variation in life satisfaction is explained by the predictors included in the model. The **Adjusted R-squared** value of **0.8415** adjusts for the number of predictors in the model, further confirming the model's good fit. The **F-statistic** (1490 on 7 and 1956 degrees of freedom) with a p-value < 2.2e-16 suggests that the model as a whole is highly significant.

### Conclusion

The analysis indicates that **GDP per capita**, **social support**, **healthy life expectancy**, **freedom to make life choices**, **generosity**, **perceptions of corruption**, and **positive affect** are significant predictors of life satisfaction. The model explains a large proportion of the variance in life satisfaction (approximately 84%), and all the predictors have a statistically significant relationship with the outcome variable.


## Project Summary

In this project, an analysis of happiness data across different countries was conducted, focusing on identifying key variables that influence people's happiness levels. The goal was to build a linear regression model to understand the factors affecting happiness and their relative importance.

### Steps Taken:
1. **Data Preprocessing** – Missing values were handled, and appropriate transformations were applied to the variables.
2. **Linear Regression Model Construction** – A linear regression model was built using selected variables to estimate the influence of factors such as GDP per capita, social support, life expectancy, and corruption on happiness levels.
3. **Diagnostic Analysis** – Diagnostic plots were created to assess model assumptions, such as normality of residuals and linearity.
4. **Results Interpretation** – The results were interpreted, identifying the most significant variables influencing happiness and evaluating model quality using R-squared and p-values.

### Findings:
The model shows that the most significant factors affecting happiness are GDP per capita, social support, and life expectancy. Additionally, perceptions of corruption have a negative impact on happiness, while positive affect and freedom to make life choices positively influence happiness.

### Final Assessment:
The project was successfully completed, and the built linear regression model provided valuable insights. The analysis met its objectives and can be used for further research or policy recommendations.
